### PR TITLE
COMMERCE-11876 make it possible to update sku expiration date if there isn't one already

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
https://liferay.atlassian.net/browse/COMMERCE-11876

resend of https://github.com/liferay-commerce/liferay-portal/pull/3956

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuResourceImpl.java [hunk 1]
```diff
@@ -406,28 +406,14 @@ private Sku _updateSKU(CPInstance cpInstance, Sku sku) throws Exception {
...
-							contextUser.getTimeZone()),
-						cpInstance.getExpirationDate())));
-
-			expirationDateMonth = expirationDateConfig.getMonth();
-			expirationDateDay = expirationDateConfig.getDay();
-			expirationDateYear = expirationDateConfig.getYear();
-			expirationDateHour = expirationDateConfig.getHour();
-			expirationDateMinute = expirationDateConfig.getMinute();
-		}
+		DateConfig expirationDateConfig = DateConfig.toExpirationDateConfig(
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuResourceImpl.java:409]: What if both `sku.getExpirationDate()` and `cpInstance.getExpirationDate()` are `null`?

Comment (victorcdp) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuResourceImpl.java:409]: @alessiorendina  That's why in the original PR, i changed the if check on line 415 to (sku.getExpirationDate() != null).
@andrea-ale-sbarra Should we put the if condition back?

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuResourceImpl.java:409]: if both are null it is set the now date +1 month. basically it's applying the same pattern of the add.

**Comment by 22LucaGiannotta22** [comment]
ci:test:commerce-functional

**Comment by 22LucaGiannotta22** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
Relevant failures are not related to the PR. It can be force forwarded

**Comment by victoroscuro** [comment]
ci:forward:force

**Pull request closed without merging**